src/main/java/com/rocketden/main/service/ProblemService.java

```diff
@@ -104,7 +105,9 @@ public ProblemDto editProblem(String problemId, ProblemDto updatedProblem) {
                 || updatedProblem.getOutputType() == null) {
             throw new ApiException(ProblemError.EMPTY_FIELD);
         }
-
+        if (updatedProblem.getApproval() && updatedProblem.getTestCases().size() == 0) {
+            throw new ApiException(ProblemError.BAD_APPROVAL);
+        }
```

Comment:
very small nit: Add lines around this if statement to separate them from the above for and below if statements.